### PR TITLE
Top-down layout child area calculation bug fix

### PR DESCRIPTION
**Pull request by Eddykasp**
The CHILD_AREA_WIDTH and CHILD_AREA_HEIGHT properties serve to specify how much space a layout requires. This is used by Topdown layout to calculate the required scale factor. The padding around the layout is not part of the layout space as its size depends on the containing node i.e. the padding should not be scaled down with the content, therefore the padding is excluded in this calculation. The three algorithms in this PR (Force, MrTree and Radial) all previously set this property incorrectly before i.e. they included the padding. This PR fixes that issue so that the scale factor is now computed correctly for these algorithms.

### CODE REVIEW

**Comment by soerendomroes** [comment]
COuld you add a description of what this actually does and maybe add a test for this?

File: plugins/org.eclipse.elk.alg.force/src/org/eclipse/elk/alg/force/ElkGraphImporter.java [hunk 1]
```diff
@@ -228,8 +228,8 @@ public void applyLayout(final FGraph fgraph) {
         if (!kgraph.getProperty(CoreOptions.NODE_SIZE_FIXED_GRAPH_SIZE)) {
             ElkUtil.resizeNode(kgraph, width, height, false, true);
         }
-        kgraph.setProperty(CoreOptions.CHILD_AREA_WIDTH, width);
-        kgraph.setProperty(CoreOptions.CHILD_AREA_HEIGHT, height);
+        kgraph.setProperty(CoreOptions.CHILD_AREA_WIDTH, width - padding.getHorizontal());
```
Comment (soerendomroes) [inline plugins/org.eclipse.elk.alg.force/src/org/eclipse/elk/alg/force/ElkGraphImporter.java:231]: This does also seem to affect the not top-down case. Have you tested this?

File: plugins/org.eclipse.elk.alg.mrtree/src/org/eclipse/elk/alg/mrtree/ElkGraphImporter.java [hunk 2]
```diff
@@ -192,8 +192,8 @@ public void applyLayout(final TGraph tGraph) {
         if (!elkgraph.getProperty(CoreOptions.NODE_SIZE_FIXED_GRAPH_SIZE)) {
             ElkUtil.resizeNode(elkgraph, width, height, false, false);
         }
-        elkgraph.setProperty(CoreOptions.CHILD_AREA_WIDTH, width);
-        elkgraph.setProperty(CoreOptions.CHILD_AREA_HEIGHT, height);
+        elkgraph.setProperty(CoreOptions.CHILD_AREA_WIDTH, width - padding.getHorizontal());
```
Comment (soerendomroes) [inline plugins/org.eclipse.elk.alg.mrtree/src/org/eclipse/elk/alg/mrtree/ElkGraphImporter.java:195]: See other comment

File: plugins/org.eclipse.elk.alg.radial/src/org/eclipse/elk/alg/radial/intermediate/CalculateGraphSize.java [hunk 3]
```diff
@@ -63,8 +63,8 @@ public void process(final ElkNode graph, final IElkProgressMonitor progressMonit
             graph.setWidth(width);
             graph.setHeight(height);
         }
-        graph.setProperty(CoreOptions.CHILD_AREA_WIDTH, width);
-        graph.setProperty(CoreOptions.CHILD_AREA_HEIGHT, height);
+        graph.setProperty(CoreOptions.CHILD_AREA_WIDTH, width - padding.getHorizontal());
```
Comment (soerendomroes) [inline plugins/org.eclipse.elk.alg.radial/src/org/eclipse/elk/alg/radial/intermediate/CalculateGraphSize.java:66]: See other comment

**Review by soerendomroes — CHANGES_REQUESTED** [review]
Why is this change only done for force, radial and mrtree and not for rectpacking and other algorithms?

File: plugins/org.eclipse.elk.alg.force/src/org/eclipse/elk/alg/force/ElkGraphImporter.java (hunk 1 above)
Comment (Eddykasp) [reply plugins/org.eclipse.elk.alg.force/src/org/eclipse/elk/alg/force/ElkGraphImporter.java:231]: No, currently this property is only used by top-down layout. In general, the child area should not include the padding by definition.

**Comment by Eddykasp** [comment]
> Why is this change only done for force, radial and mrtree and not for rectpacking and other algorithms?

Because the property is only set incorrectly for these three algorithms.

**Pull request merged**